agents/PCS-Large-LP-Deposit-Withdrawal/src/agent.ts

```diff
@@ -0,0 +1,52 @@
+import { BigNumber } from "ethers";
+import { Finding, HandleTransaction, TransactionEvent, LogDescription, getEthersProvider } from "forta-agent";
+import { EVENTS_ABI, POOL_SUPPLY_THRESHOLD, THRESHOLD_PERCENTAGE, FACTORY } from "./constants";
+import { createPair } from "./utils";
+import { createFinding } from "./finding";
+import PoolFetcher from "./pool.fetcher";
+const STKBNBPOOLADDRESS = "0xaa2527ff1893e0d40d4a454623d362b79e8bb7f1";
+
+export const provideHandleTransaction =
+  (
+    createPair: any,
+    fetcher: PoolFetcher,
+    poolSupplyThreshold: BigNumber,
+    thresholdPercentage: BigNumber,
+    factory: string
+  ): HandleTransaction =>
+  async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const block: number = txEvent.blockNumber;
+    const logs: LogDescription[] = txEvent.filterLog(EVENTS_ABI);
+    if (!logs) return findings;
+    await Promise.all(
+      logs.map(async (log) => {
+        const { amount0, amount1 } = log.args;
+        const [valid, token0, token1, totalSupply] = await fetcher.getPoolData(block - 1, log.address);
+        const createdPair = createPair(token0, token1, factory);
```

Comment:
As far as I see, token0 & token1 are constants for a fixed pool. So, I don't think you need to query them everytime. You can jt find them and declare them as constants within the code. The pool is never going to change for us, its fixed stkBNB-BNB pool. So, you should take advantage of that and optimize your code.

Reply:
Done